### DISCUSSION TITLE
Environmental variables with Vite?

### DISCUSSION THREAD

**Discussion by jameschetwood** [crxjs/chrome-extension-tools#774] [Q&A]
Can you send environmental variables via different npm scripts in Vite?

I was hoping to do something like this: 
```
  "scripts": {
    "dev": "vite",
    "dev-marketing": "VITE_MARKETING=yes vite",
```

**Comment by JoeyDoey** [comment]
I think you can just use [Vite's `mode`](https://vitejs.dev/guide/env-and-mode.html#modes)

Update your package.json script to e.g.

```json
 "dev-marketing": "vite --mode marketing"
```
you should now have that "mode === marketing" when you run Vite build/dev

```typescript
export default defineConfig(({ command, mode }) => {
  console.log({ command, mode }) 
  //   👆 can log to test output. 
  // fyi, `command` will be either dev or build (serve/build)
  // in case you have different logic when building vs when just dev-ing
})
```

hope that helps! 🚀

ps: if you _need_ to use env vars, use something like [cross-env](https://www.npmjs.com/package/cross-env). I think. you should be able to do something like this (not tested):

```json
  "dev-marketing": "cross-env VITE_MARKETING=yes vite"
```

- - - - 
ref:
- [Vite Env Vars & Modes](https://vitejs.dev/guide/env-and-mode.html#modes)
- [cross-env (npm)](https://www.npmjs.com/package/cross-env)
